lib/rucio/core/did.py

```diff
@@ -1829,3 +1830,115 @@ def list_archive_content(scope, name, session=None):
                    'bytes': tmp_did.bytes, 'adler32': tmp_did.adler32, 'md5': tmp_did.md5}
     except NoResultFound:
         raise exception.DataIdentifierNotFound("Data identifier '%(scope)s:%(name)s' not found" % locals())
+
+
+@transactional_session
+def add_did_to_followed(scope, name, account, session=None):
+    """
+    Mark a did as followed by the given account
+
+    :param scope: The scope name.
+    :param name: The data identifier name.
+    :param account: The account owner.
+    :param session: The database session in use.
+    """
+    return add_dids_to_followed(dids=[{'scope': scope, 'name': name}],
+                                account=account, session=session)
+
+
+@transactional_session
+def add_dids_to_followed(dids, account, session=None):
+    """
+    Bulk mark datasets as followed
+
+    :param dids: A list of dids.
+    :param account: The account owner.
+    :param session: The database session in use.
+    """
+    try:
+        for did in dids:
+            did = session.query(models.DataIdentifier).filter_by(scope=did['scope'], name=did['name']).one()
+            new_did_followed = models.DidsFollowed(scope=did.scope, name=did.name, account=account,
+                                                   did_type=did.did_type, expired_at=did.expired_at)
+
+            new_did_followed.save(session=session, flush=True)
```

Comment:
any reason why you flush to the DB every iteration? just curious

Reply:
No Sir, not any specific reason as such. I will make the changes and flush after the loop is completed.